arctic/store/_ndarray_store.py

```diff
@@ -163,6 +157,47 @@ def get_info(self, version):
         ret['rows'] = int(version['up_to'])
         return ret
 
+    def chunked_read(self, arctic_lib, version, symbol, date_range, read_preference=None):
+        collection = arctic_lib.get_top_level_collection()
+        if read_preference:
+            collection = collection.with_options(read_preference=read_preference)
+        return self._do_chunked_read(collection, version, symbol, date_range)
+
+    def _do_chunked_read(self, collection, version, symbol, date_range):
+        segment_count = version.get('segment_count', None)
+
+        spec = {'symbol': symbol,
+                'parent': version.get('base_version_id', version['_id']),
+                }
+
+        if date_range is not None:
+            spec['segment'] = date_range.mongo_query()
+
+        segments = []
+        i = -1
+        for i, x in enumerate(collection.find(spec, sort=[('segment', pymongo.ASCENDING)],)):
+            try:
+                segments.append(decompress(x['data']))
+            except Exception:
+                dump_bad_documents(x,
```

Comment:
probably not. It doesnt seem like we need it in the normal read path either, but I wasnt sure, so I just left it in. Do you think we can remove this from both paths @richardbounds ?